src/crypto.rs

```diff
@@ -30,11 +30,14 @@ fn generichash_quick(msg: &[u8], key: Option<&[u8]>) -> Result<[u8; 32]> {
         .expect("generichash returned result of wrong size"))
 }
 
-pub fn init() -> Result<()> {
+pub(crate) fn init() -> Result<()> {
     to_enc_error!(sodiumoxide::init(), "Failed initialising libsodium")
 }
 
-pub fn derive_key(salt: &[u8; SALT_SIZE], password: &str) -> Result<[u8; SYMMETRIC_KEY_SIZE]> {
+pub(crate) fn derive_key(
```

Comment:
This one we actually want public, people may want to derive their own keys, it's useful.

Reply:
It was only exported through `test_helpers`, so does `test_helpers` have to be considered part of the public API?

Reply:
Oh, maybe we didn't expose it in the end then. I thought we have when we started supporting login by key (rather than password).